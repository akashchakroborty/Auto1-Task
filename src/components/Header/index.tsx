import React from 'react';
import Logo from '../../assets/logo.png';
import { makeStyles, createStyles } from '@mui/styles';
import { Theme, AppBar, Toolbar, Link } from '@mui/material';
import { NavLink } from 'react-router-dom';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    appBar: {
      borderBottom: `1px solid ${theme.palette.divider}`,
    },
    logo: {
      margin: '10px 0',
    },
  }),
);

const Header = () => {
  const classes = useStyles();

  return (
    <AppBar position="static" color="inherit" elevation={0} className={classes.appBar}>
      <Toolbar>
        <NavLink to="/" style={{ flexGrow: 1 }}>
          <img src={Logo} alt="Auto1 Logo" className={classes.logo} />
        </NavLink>
        <nav>
          <Link variant="button" color="textPrimary" component={NavLink} to="/purchase">
            Purchase
          </Link>
          <Link variant="button" color="textPrimary" component={NavLink} to="/orders">
            My Orders
          </Link>
          <Link variant="button" color="textPrimary" component={NavLink} to="/sell">
            Sell
          </Link>
        </nav>
      </Toolbar>
    </AppBar>
  );
};

export default Header;
