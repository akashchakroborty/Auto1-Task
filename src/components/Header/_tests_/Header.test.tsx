/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import Header from '..';

jest.mock('@mui/styles', () => {
  const Styles = jest.requireActual('@mui/styles');
  const theme = { palette: { divider: {} } };

  return {
    ...Styles,
    makeStyles: (func: any) => {
      return Styles.makeStyles(func.bind(null, theme));
    },
  };
});

describe('should test the Header Component', () => {
  test('should render the component', () => {
    const component = shallow(<Header />);
    expect(component).toMatchSnapshot();
  });
});
