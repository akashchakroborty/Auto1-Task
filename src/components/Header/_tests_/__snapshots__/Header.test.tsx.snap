// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test the Header Component should render the component 1`] = `
<ForwardRef(AppBar)
  className="makeStyles-appBar-1"
  color="inherit"
  elevation={0}
  position="static"
>
  <ForwardRef(Toolbar)>
    <NavLink
      style={
        Object {
          "flexGrow": 1,
        }
      }
      to="/"
    >
      <img
        alt="Auto1 Logo"
        className="makeStyles-logo-3"
        src=""
      />
    </NavLink>
    <nav>
      <ForwardRef(Link)
        className="makeStyles-link-2"
        color="textPrimary"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "activeClassName": [Function],
              "activeStyle": [Function],
              "aria-current": [Function],
              "className": [Function],
              "exact": [Function],
              "innerRef": [Function],
              "isActive": [Function],
              "location": [Function],
              "onClick": [Function],
              "replace": [Function],
              "sensitive": [Function],
              "strict": [Function],
              "style": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        to="/purchase"
        variant="button"
      >
        Purchase
      </ForwardRef(Link)>
      <ForwardRef(Link)
        className="makeStyles-link-2"
        color="textPrimary"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "activeClassName": [Function],
              "activeStyle": [Function],
              "aria-current": [Function],
              "className": [Function],
              "exact": [Function],
              "innerRef": [Function],
              "isActive": [Function],
              "location": [Function],
              "onClick": [Function],
              "replace": [Function],
              "sensitive": [Function],
              "strict": [Function],
              "style": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        to="/orders"
        variant="button"
      >
        My Orders
      </ForwardRef(Link)>
      <ForwardRef(Link)
        className="makeStyles-link-2"
        color="textPrimary"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "activeClassName": [Function],
              "activeStyle": [Function],
              "aria-current": [Function],
              "className": [Function],
              "exact": [Function],
              "innerRef": [Function],
              "isActive": [Function],
              "location": [Function],
              "onClick": [Function],
              "replace": [Function],
              "sensitive": [Function],
              "strict": [Function],
              "style": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        to="/sell"
        variant="button"
      >
        Sell
      </ForwardRef(Link)>
    </nav>
  </ForwardRef(Toolbar)>
</ForwardRef(AppBar)>
`;
